File: week2/c++/week2_2.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> prices)
{
    vector<int> answer(prices.size());

    for (int i = 0; i < prices.size(); i++)
    {
        for (int j = i + 1; j < prices.size(); j++)
        {
            // 증가
            if (prices[i] <= prices[j])
            {
                answer[i]++;
            }
            // 감소
            else
            {
                answer[i]++;
                break;
            }
        }
    }

    return answer;
}

File: week2/java/week2_2.java
public class week2_2 {
    public static int[] solution(int[] prices){
        int[] answer = new int[prices.length];

        for(int i=0; i<prices.length; i++){
            int cnt = 0;
            for(int j=i+1; j<prices.length; j++){
                cnt++;
                if(prices[i] > prices[j]) break;
            }
            answer[i] = cnt;
        }

        return answer;
    }
    public static void main(String[] args){

    }
}